src/core/utils/positioningutils.cpp

```diff
@@ -0,0 +1,113 @@
+  double hdop = 0;
+  double vdop = 0;
+  double hacc = std::numeric_limits<double>::quiet_NaN();
+  double vacc = std::numeric_limits<double>::quiet_NaN();
+  double hvacc = std::numeric_limits<double>::quiet_NaN();
+  double verticalSpeed = std::numeric_limits<double>::quiet_NaN();
+  double magneticVariation = std::numeric_limits<double>::quiet_NaN();
+
+  QDateTime utcDateTime = positionsInformation.last().utcDateTime();
+
+  QList<QgsSatelliteInfo> satellitesInView = positionsInformation.at( 0 ).satellitesInView();
+  int satellitesUsed = satellitesInView.size();
+  bool satInfoComplete = positionsInformation.at( 0 ).satInfoComplete();
+  QList<int> satPrn = positionsInformation.at( 0 ).satPrn();
+  QChar status = positionsInformation.at( 0 ).status();
+  QChar fixMode = positionsInformation.at( 0 ).fixMode();
+  int fixType = positionsInformation.at( 0 ).fixType();
+  int quality = positionsInformation.at( 0 ).quality();
+  QString sourceName = positionsInformation.at( 0 ).sourceName();
+
+  for ( const GnssPositionInformation &pi : positionsInformation )
+  {
+    if ( !std::isnan( pi.latitude() ) )
+      latitude = !std::isnan( latitude ) ? latitude + pi.latitude() : pi.latitude();
+    if ( !std::isnan( pi.longitude() ) )
+      longitude = !std::isnan( longitude ) ? longitude + pi.longitude() : pi.longitude();
+    if ( !std::isnan( pi.elevation() ) )
+      elevation = !std::isnan( elevation ) ? elevation + pi.elevation() : pi.elevation();
+    if ( !std::isnan( pi.speed() ) )
+      speed = !std::isnan( speed ) ? speed + pi.speed() : pi.speed();
+    if ( !std::isnan( pi.direction() ) )
+      direction = !std::isnan( direction ) ? direction + pi.direction() : pi.direction();
+    pdop += pi.pdop();
+    hdop += pi.pdop();
+    vdop += pi.pdop();
+    if ( !std::isnan( pi.hacc() ) )
+      hacc = !std::isnan( hacc ) ? hacc + pi.hacc() : pi.hacc();
+    if ( !std::isnan( pi.vacc() ) )
+      vacc = !std::isnan( vacc ) ? vacc + pi.vacc() : pi.vacc();
+    if ( !std::isnan( pi.hvacc() ) )
+      hvacc = !std::isnan( hvacc ) ? hvacc + pi.hvacc() : pi.hvacc();
+    if ( !std::isnan( pi.verticalSpeed() ) )
+      verticalSpeed = !std::isnan( verticalSpeed ) ? verticalSpeed + pi.verticalSpeed() : pi.verticalSpeed();
+    if ( !std::isnan( pi.magneticVariation() ) )
+      magneticVariation = !std::isnan( magneticVariation ) ? magneticVariation + pi.magneticVariation() : pi.magneticVariation();
+  }
+
+  return GnssPositionInformation( latitude / positionsInformationCount, longitude / positionsInformationCount, elevation / positionsInformationCount,
+                                  speed / positionsInformationCount, direction / positionsInformationCount, satellitesInView,
+                                  pdop / positionsInformationCount, hdop / positionsInformationCount, vdop / positionsInformationCount,
+                                  hacc / positionsInformationCount, vacc / positionsInformationCount, utcDateTime,
```

Comment:
I think we cannot average acc values.
Not sure what's best. Lowest? Set to nan, 0?

Reply:
@m-kuhn , isn't an averaged set of accuracy values better than nan or 0? Understanding that it doesn't inherently have the same meaning as a single position's accuracy, it still tells you something about the overall accuracy of the averaged points.

Reply:
Good point, convinced.
Let's make that transparent in docs.